### PR TITLE
implement observer pattern for collisions

### PR DESCRIPTION
**Pull request by pfent**
currently handled with a collisionreason, and currently notifies a FruitRNG, the Board and the Snake

### CODE REVIEW

File: src/main/java/nl/tudelft/tud15a/snake/model/FruitRNG.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        Fruit fruit;
+        if(Math.random() > 0.8) {
+            fruit = new Golden(new Apple());
+        } else {
+            fruit = new Apple();
+        }
+        model.setFruit(fruit);
+        fruit.locate();
+    }
+}
```
Comment (canbilal1995) [inline src/main/java/nl/tudelft/tud15a/snake/model/FruitRNG.java:27]: this part is ok

File: src/main/java/nl/tudelft/tud15a/snake/model/observer/CollisionListener.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package nl.tudelft.tud15a.snake.model.observer;
+
+public interface CollisionListener {
+    void onCollision(CollisionReason reason);
+}
```
Comment (canbilal1995) [inline src/main/java/nl/tudelft/tud15a/snake/model/observer/CollisionListener.java:5]: I thinkpretty good observer pattern

**Review by canbilal1995 — COMMENTED** [review]
It's OK for me. What about others?

**Review by jguinea — APPROVED** [review]
Great job!

**Review by canbilal1995 — APPROVED** [review]

**Pull request merged**